### ISSUE TITLE
Pipes Chapter sample code doesn't compile

### ISSUE THREAD

**Issue by johnbiundo** [nestjs/docs.nestjs.com#366]
## I'm submitting a...

<pre><code>
[ ] Regression 
[X] Bug report
[ ] Feature request
[ ] Documentation issue or request (new chapter/page)
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Use the code from the Class validator example (validation.pipe.ts).

```
import { PipeTransform, Injectable, ArgumentMetadata, BadRequestException } from '@nestjs/common';
import { validate } from 'class-validator';
import { plainToClass } from 'class-transformer';

@Injectable()
export class ValidationPipe implements PipeTransform<any> {
  async transform(value: any, { metatype }: ArgumentMetadata) {
    if (!metatype || !this.toValidate(metatype)) {
      return value;
    }
    const object = plainToClass(metatype, value);
    const errors = await validate(object);
    if (errors.length > 0) {
      throw new BadRequestException('Validation failed');
    }
    return value;
  }

  private toValidate(metatype: Function): boolean {
    const types = [String, Boolean, Number, Array, Object];
    return !types.includes(metatype);
  }
}
```

It fails to compile with this error: 

```
TSError: Unable to compile TypeScript:
src/common/pipes/validation.pipe.ts(26,28): error TS2345: Argument of type 'Function' is not assignable to parameter of type 'StringConstructor | BooleanConstructor | NumberConstructor | ArrayConstructor | ObjectConstructor'.
  
Type 'Function' is missing the following properties from type 'ObjectConstructor': getPrototypeOf, getOwnPropertyDescriptor, getOwnPropertyNames, create, and 16 more.
```

Examining the [built-in Validation Pipe](https://github.com/nestjs/nest/blob/master/packages/common/pipes/validation.pipe.ts) and how it handles this, I think the following lines in the example:
```
  private toValidate(metatype: Function): boolean {
    const types = [String, Boolean, Number, Array, Object];
    return !types.includes(metatype);
  }
```

could be replaced by these lines:
```
  private toValidate(metatype): boolean {
    const types = [String, Boolean, Number, Array, Object];
    return !types.some(t => metatype === t);
  }
```

I have done some minimal testing and this appears to work, though I'm not quite strong enough with TypeScript yet to be 100% sure this is correct.

If I can get some consensus on this change, I'll include it with a PR for some other changes to the Pipes chapter.

## Expected behavior

It should compile and run

## Environment

<pre><code>
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Hey @johnbiundo,
you can just explicitly cast `types` array:
```typescript
// before
private toValidate(metatype: Function): boolean {
  const types = [String, Boolean, Number, Array, Object];
  return !types.includes(metatype);
}
// after
private toValidate(metatype: Function): boolean {
  const types: Function[] = [String, Boolean, Number, Array, Object];
  return !types.includes(metatype);
}
```
Thanks for reporting btw!

**Comment by johnbiundo** [comment]
Oh, yeah, that's better :-)

Tested and working.  I'll update the PR with this change.

Thanks.

**Comment by johnbiundo** [comment]
Fixed by PR #368
